llvm/cmake/modules/AddLLVM.cmake

```diff
@@ -636,10 +636,18 @@ function(llvm_add_library name)
   endif()
 
   if(ARG_MODULE)
-    add_library(${name} MODULE ${ALL_FILES})
+    if(EMSCRIPTEN)
+      add_library(${name} STATIC ${ALL_FILES})
+    else()
+      add_library(${name} MODULE ${ALL_FILES})
+    endif()
   elseif(ARG_SHARED)
     add_windows_version_resource_file(ALL_FILES ${ALL_FILES})
-    add_library(${name} SHARED ${ALL_FILES})
+    if(EMSCRIPTEN)
+      add_library(${name} STATIC ${ALL_FILES})
+    else()
+      add_library(${name} SHARED ${ALL_FILES})
+    endif()
```

Comment:
Would it make more sense to set `ARG_SHARED` and `ARG_MODULE` to false above?

Reply:
Yes I think that would do to !

Reply:
Made the change.